modules/phase_field/include/userobjects/FauxPolycrystalVoronoi.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "PolycrystalVoronoi.h"
+
+// Forward Declarations
+class FauxPolycrystalVoronoi;
+
+template <>
+InputParameters validParams<FauxPolycrystalVoronoi>();
```

Comment:
Use the `static` class method instead. The function template specialization is deprecated